### PR TITLE
Refactor device detection, allow override

### LINKED ISSUE #1203: unreliable scsi symlinks
### Description of Issue/Question
With #952 and #1061, DeepSea started selecting other symlinks.  However, the udev rules from 58-scsi-sg3_symlink.rules do not create persistent and unique names.  On hardware with these devices, zapping an OSD causes the device name to be given to another device.

"I obviously cannot choose the wine in front of you" - Vizzini, Princess Bride

After examining 58-scsi-sg3_symlink.rules and 60-persistent-storage.rules, I believe there is no reason to use any of the `scsi-` prefixed names.  The most reliable symlink would be the `scsi-3` prefixed devices but these are redundant with `wwn-0x` devices.

Rather than have complicated logic including or excluding particular expressions of devices, removing the `scsi-` prefix entirely will simplify the entire process.  For any hardware that has an `ata-` prefix, that will get chosen.  Otherwise, the `wwn` will be selected.

### PR DESCRIPTION
**Pull request by swiftgist**
Signed-off-by: Eric Jackson <ejackson@suse.com>

Fixes #1203 
bsc: 1097914

### CODE REVIEW

File: srv/salt/_modules/cephdisks.py [hunk 1]
```diff
@@ -504,6 +504,81 @@ def assemble_device_list(self):
...
+    else:
+        pathname = _seek('ceph:modules:cephdisks:device:pathname'.split(':'), __pillar__)
+        if pathname:
+            log.warning("Using pillar value {} for pathname".format(pathname))
+        else:
+            pathname = "/dev/disk/by-id"
+    return pathname
+
+
+def _seek(keys, saltdict):
```
Comment (jan--f) [inline srv/salt/_modules/cephdisks.py]: I think I'd be more comfortable with a non-recursive seek. E.g.
```
def _seek(keys, dict)
  tmp = dict
  for key in keys:
    if key in tmp:
      tmp = tmp[key]
    else:
      return None
  return tmp
```
Personally I find that easier to read...though I suppose thats nit picking.

**Comment by jschmid1** [comment]
This PR is a bit dependent on #1220's outcome ( or the way around )

**Comment by jschmid1** [comment]
@swiftgist rebase please

**Comment by swiftgist** [comment]
done

**Comment by jschmid1** [comment]
the comment from here: https://github.com/SUSE/DeepSea/issues/1203#issuecomment-405557398

should go in the wiki/documentation

**Comment by jschmid1** [comment]
@swiftgist ^^^

**Comment by kshtsk** [comment]
retest this please

**Comment by kshtsk** [comment]
retest this please

**Comment by jschmid1** [comment]
[Replace/Migration Pullrequests](https://trello.com/c/TBLiA9cD/9-replace-migration-pullrequests)

**Comment by jschmid1** [comment]
@swiftgist we need a backport of this

**Comment by jschmid1** [comment]
@swiftgist could you please rebase

**Comment by swiftgist** [comment]
rebased to include newer tests.

**Comment by jschmid1** [comment]
@susebot run teuthology

**Comment by denisok** [comment]
@susebot run teuthology

**Comment by denisok** [comment]
@susebot run teuthology

**Comment by jschmid1** [comment]
huh @denisok teuthology died here

**Comment by jschmid1** [comment]
@susebot run teuthology

**Pull request merged**